src/ax/endpoint/validator.go

```diff
@@ -0,0 +1,47 @@
+package endpoint
+
+import (
+	"context"
+
+	"github.com/jmalloc/ax/src/ax"
+)
+
+// Validator wraps methods required to
+// perform message validation
+type Validator interface {
+	Validate(ctx context.Context, msg ax.Message) error
+}
+
+// DefaultValidators is the array of validators used by default
+// when sending a message in case none of user-defined validators
+// have been defined.
+var DefaultValidators = []Validator{
+	&BasicValidator{},
+}
+
+// BasicValidator is one of the default message validators
+// that performs basic checks on the message such as
+// if a message is nil, etc.
+type BasicValidator struct{}
```

Comment:
I think the `nil` check can go.  When we talked about that originally, I was imagining that the validator would get passed the `Envelope`, and it would check that `env.Message` was non-nil, but I actually prefer the way you've done it here - the application shouldn't be concerned with validating the envelope itself.

With the `nil` check gone, this implementation would only be concerned with the `SelfValidatingMessage` interface, so it could be called a `SelfValidator`  or something to that effect so it's explicit about what it's for. I don't love the name `SelfValidator`, so open to better names there for sure.

Reply:
Renamed go `SelfValidator` for now.